### PR TITLE
SVC-100: Configure relay class to store temporary copies of relayed logs

### PR DESCRIPTION
**Pull request by billglick**
See last few comments of https://jira.ncsa.illinois.edu/browse/SVC-100

This is being tested on the asd-log host.

### CODE REVIEW

**Review by andylytical — COMMENTED** [review]
The word `remote` is quite ambiguous. Looks like it is used for both incoming and outgoing messages (took me three reads to figure that out). Can the variable names and the Ruleset names be changed to be more self-descriptive?

I'm also confused if `local` means "messages generated from the localhost" or "messages that are destined to remain on the localhost but may have been received from somewhere else". Does it make sense to also adjust these to be more descriptive?

**Comment by billglick** [comment]
@andylytical ,

What about using these phrases for descriptive names?
- `localhost` to denote actions, rules, templates, etc. applying to log messages from the local host
- `collected` to denote actions, rules, templates, etc applying to log messages collected/sent/relayed from another host
- `forwarded` to denote actions, rules, etc applying to log messages that are forwarded to another host

These are only used in the examples within the README.md file (and copied into control repos), but I agree that anything that we can do to make that more readable and understandable would be very helpful.

This perhaps also suggests to rename the `profile_rsyslog::relay` class to `profile_rsyslog::collector`. Thoughts?

**Comment by andylytical** [comment]
@billglick I like all those naming suggestions. Let's go with those.

Regarding renaming the class, also a good idea.

How much trouble do you think it would be to rename the profile? It doesn't match our standard of "name by purpose, not implementation". Possible names are `profile::syslog` or `profile::systemlogs`. Thoughts?

**Comment by billglick** [comment]
@andylytical Sounds good. 

Renaming the profile wouldn't be hard, but I'm worried about breaking current things. How about we delay that to another ticket/issue? And we basically archive this one and replace it with a new `profile_syslog`. I agree that it is better to do it sooner than later.  I'll create a GitHub issue and Jira ticket for that.

**Comment by billglick** [comment]
@andylytical How do these updated hiera examples now read in README.md?

I'll rebase after we finish reviewing changes.

File: README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ profile_rsyslog::config_rulesets:
           type: "omfile"
           facility: "cron.*"
           config:
-            file: "/var/log/cronlog"
+            file: "/var/log/cron"
```
Comment (billglick) [inline README.md:75]: This was changed to match file that RHEL already uses.
I presumed this was accidentally set before.

Comment (andylytical) [reply README.md:75]: That reads 1000% percent better! Night and Day! What an incredible improvement.
Thanks for the extra effort, @billglick !

**Comment by billglick** [comment]
@andylytical Are you still reviewing this? Or can I proceed with merging it?

**Review by andylytical — APPROVED** [review]
Looks good. Really like the new names!

**Pull request merged**